crates/autopilot/src/solvable_orders.rs

```diff
@@ -656,10 +682,10 @@ fn to_normalized_price(price: f64) -> Option<U256> {
     }
 }
 
-async fn filter_unsupported_tokens(
-    mut orders: Vec<Order>,
+async fn find_unsupported_tokens(
+    orders: &[Order],
     bad_token: Arc<dyn BadTokenDetecting>,
-) -> Vec<Order> {
+) -> Vec<OrderUid> {
     let bad_tokens = join_all(
         orders
             .iter()
```

Comment:
Just noticed that in the line below (can't comment on that line) you can also use `token_pair().into_iter().flatten()` to avoid the weird `TokenPair::default()` implementation.